### ISSUE TITLE
Send money - Missing bank account option when paying with Expensify

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#38739]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.55-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
1. Go to staging.new.expensify.com.
2. Go to 1:1 DM
3. Click + > Send money
4. Select USD currency and enter an amount > Click Next
5. Click Pay with Expensify.

## Expected Result:
Bank account option will be present in Pay with Expensify menu.

## Actual Result:
Bank account option is not present in Pay with Expensify menu.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/dbff58e5-666f-4c5d-ae2f-6349ed714699

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
We think this issue might be related to the #vip-vsb.

**Comment by godofoutcasts94** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Send money - Missing bank account option when paying with Expensify
### What is the root cause of that problem?
Here :
https://github.com/Expensify/App/blob/51ef6c1bd1cd3d981db744bff1fcff9e7c670c0c/src/components/AddPaymentMethodMenu.js#L69-L73
we need to add `Send Report Request` as well
### What changes do you think we should make in order to solve the problem?

Here : 
https://github.com/Expensify/App/blob/51ef6c1bd1cd3d981db744bff1fcff9e7c670c0c/src/components/AddPaymentMethodMenu.js#L69-L73
we can add this `ReportUtils.isMoneyRequest` in one of the `OR` condition
### What alternative solutions did you explore? (Optional)
N/A

### Result

<img width="1470" alt="Screenshot 2024-03-21 at 4 54 12 PM" src="https://github.com/Expensify/App/assets/158435970/6cbb5c90-cab3-4da2-9c9f-ebcc846c3712">

**Comment by godofoutcasts94** [comment]
Updated [Proposal](https://github.com/Expensify/App/issues/38739#issuecomment-2012007752)

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The bank account option isn't available when sending money to a user.

### What is the root cause of that problem?
The bank account page will show if it's an expense or an IOU report.
https://github.com/Expensify/App/blob/82065804528dc66daaa6ba652ab2dea84684c619/src/components/AddPaymentMethodMenu.js#L71-L75

However, on the money request confirmation page, the expense/IOU report doesn't exist. Each send money will create a new IOU report. Also, in our case, we can only send money to a user and not a workspace, so the expected result would be to show the personal bank account option.

### What changes do you think we should make in order to solve the problem?

We need to pass `shouldShowPersonalBankAccountOption` to the confirmation page settlement button.
https://github.com/Expensify/App/blob/82065804528dc66daaa6ba652ab2dea84684c619/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js#L622-L624

The props is passed down deeper to the AddPaymentMethodMenu component.

_this is what we do in [here](https://github.com/Expensify/App/issues/32079) too_

**Comment by stephanieelliott** [comment]
Hey @jjcoffee, we have a few proposals for review when you get a chance!

**Comment by jjcoffee** [comment]
@godofoutcasts94's [proposal](https://github.com/Expensify/App/issues/38739#issuecomment-2012007752) is unclear and has an incomplete solution, which doesn't appear to work. I'd recommend working on expanding your proposals in future to make it clearer both for yourself and for the reviewers.

The other two are pretty similar proposals, but @bernhardoj's [proposal](https://github.com/Expensify/App/issues/38739#issuecomment-2012041824) is the first to get the correct RCA and solution. We don't need to check for `isTypeSend` as @GandalfGwaihir [suggests](https://github.com/Expensify/App/issues/38739#issuecomment-2012146496), as we already do that in `shouldShowSettlementButton`.

https://github.com/Expensify/App/blob/82065804528dc66daaa6ba652ab2dea84684c619/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js#L619

:ribbon::eyes::ribbon: C+ reviewed

**Comment by godofoutcasts94** [comment]
Hey @jjcoffee please have a look at the working video of my proposal

**Comment by godofoutcasts94** [comment]
> @godofoutcasts94 The expected result here is to show the personal bank account option, as the other two proposals implement. You also have `ReportUtils.isMoneyRequest` in your solution without passing anything to it, which is why I recommend digging into the details for your future proposals. Best of luck!

but in the issue description, there is no discussion about `Personal` or `Business` Bank Account Option.

**Comment by arosiclair** [comment]
@bernhardoj's propsal LGTM though I'm not sure if bank accounts are actually supposed to be an option for the Send Money flow currently. I'm asking internally [here](https://expensify.slack.com/archives/C03U7DCU4/p1711466854570889).

**Comment by arosiclair** [comment]
Alright just concluded that conversation. We _should_ list both Personal and Business bank accounts as options when using `Pay with Expensify` in the send money flow. Any of those can be linked to your wallet for payment.

That means both @bernhardoj and @godofoutcasts94's proposals are actually incomplete. Can either of you update your proposals and can you re-review @jjcoffee? I think we can take whichever is updated and acceptable first.

**Comment by godofoutcasts94** [comment]
updated [proposal](https://github.com/Expensify/App/issues/38739#issuecomment-2012007752) 

cc - @arosiclair

**Comment by godofoutcasts94** [comment]
Updated [proposal](https://github.com/Expensify/App/issues/38739#issuecomment-2012007752) with alternative solution as well.

**Comment by bernhardoj** [comment]
@arosiclair wouldn't it be redundant to show both business and personal bank account? Both options will navigate the user to add a personal bank account.

**Comment by arosiclair** [comment]
> @arosiclair wouldn't it be redundant to show both business and personal bank account? Both options will navigate the user to add a personal bank account.

I don't think this is true. The business bank account option should create a workspace and start the VBBA flow [here](https://github.com/Expensify/App/blob/6d06a30801b063ff4b7dddb8e77b8196d8761b6e/src/components/KYCWall/BaseKYCWall.tsx#L128).

**Comment by bernhardoj** [comment]
@arosiclair it will create a workspace and start the VBBA if we do it on an existing iou report, but when sending money, we are always creating a new iou report, and it's not possible to send money to workspace. So, it will navigate to `addBankAccountRoute` instead.
https://github.com/Expensify/App/blob/6d06a30801b063ff4b7dddb8e77b8196d8761b6e/src/components/KYCWall/BaseKYCWall.tsx#L137C37-L137C56

And `addBankAccountRoute` will be the add personal bank account route.
https://github.com/Expensify/App/blob/6d06a30801b063ff4b7dddb8e77b8196d8761b6e/src/libs/ReportUtils.ts#L890-L892

**Comment by arosiclair** [comment]
> @arosiclair it will create a workspace and start the VBBA if we do it on an existing iou report, but when sending money, we are always creating a new iou report, and it's not possible to send money to workspace. So, it will navigate to `addBankAccountRoute` instead.

Hmm that sounds like another bug then. I would expect a workspace to be created in both cases. 

Alright, let's just focus on just the personal bank account in this issue and then I'll open a follow up to give special attention to the send money flow using a business bank account. Since @bernhardoj had the best proposal for the personal bank account, I'll hire them.

**Comment by bernhardoj** [comment]
PR is ready

cc: @jjcoffee

**Comment by stephanieelliott** [comment]
PR has been reviewed, currently undergoing QA

**Comment by stephanieelliott** [comment]
PR is on staging

**Comment by stephanieelliott** [comment]
Reapplying the `Bug` label to get another BZ member on this while I am OOO til May 2. Thanks @zanyrenney,  only thing to do here is issue payment on 4/25!

**Comment by zanyrenney** [comment]
**payment summary**

@jjcoffee PAID $500 ON UW [automatic offer](https://www.upwork.com/nx/wm/offer/0) (Reviewer)
@bernhardoj  PAID $500 ON UW [automatic offer](https://www.upwork.com/nx/wm/offer/0) (Contributor)
